runtime/interpreter/decode.go

```diff
@@ -1736,3 +1625,186 @@ func (d *DecoderV4) decodeCapabilityStaticType() (StaticType, error) {
 		BorrowType: borrowStaticType,
 	}, nil
 }
+
+// decodeCompositeMetaInfo decodes the meta info from the byte content and updates the composite value.
+// Meta info includes:
+//    - location
+//    - qualifiedIdentifier
+//    - kind
+//
+// This also extracts out the fields raw content and cache it separately inside the value.
+//
+func decodeCompositeMetaInfo(v *CompositeValue, content []byte) error {
+	d, err := NewByteDecoder(content, v.Owner, CurrentEncodingVersion, v.decodeCallback)
```

Comment:
The version of the data might not be the current version. The version of the encoded data should be saved in the deferred composite and then used here